api/uaa/auth.go

```diff
@@ -18,17 +19,23 @@ type AuthResponse struct {
 
 // Authenticate sends a username and password to UAA then returns an access
 // token and a refresh token.
-func (client Client) Authenticate(ID string, secret string, grantType constant.GrantType) (string, string, error) {
+func (client Client) Authenticate(ID string, secret string, origin string, grantType constant.GrantType) (string, string, error) {
 	requestBody := url.Values{
 		"grant_type": {string(grantType)},
 	}
 	switch grantType {
 	case constant.GrantTypeClientCredentials:
 		requestBody.Set("client_id", ID)
 		requestBody.Set("client_secret", secret)
+		if origin != "" {
+			return "", "", errors.New("Incorrect Usage: The following arguments cannot be used together: --client-credentials, --origin")
```

Comment:
This error contains flag information. The API clients _should_ know nothing about the command line.

Move this validation into the command layer. Look at [create-buildpack's `--enable`/`--disable`](https://github.com/cloudfoundry/cli/blob/master/command/v2/create_buildpack_command.go#L66-L70) to see how we implement it. And don't forget to add command layer tests for them.